src/hooks/useOS.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from "react";
+
+export type OS = "mac" | "other";
+
+export const useOS = () => {
+  const [os, setOs] = useState<OS>("mac");
```

Comment:
Handling this doesn't require a state, actually! 

here's the revised code
```
export type OS = 'mac' | 'windows' | 'linux' | 'other';

export const useOS = (): OS => {
  const getOperatingSystem = (): OS => {
    const platform = window.navigator.platform.toLowerCase();

    if (platform.includes('mac')) return 'mac';
    if (platform.includes('win')) return 'windows';
    if (platform.includes('linux')) return 'linux';
    
    return 'other';
  };

  return useMemo(() => getOperatingSystem(), []);
};
```

other platforms are not needed, so it's ok to just use `mac` and `other` in this case, but the `useMemo` is an important part. It'll run only once and won't cause the rerender of component using this hook, as it's the case in an option with using state.

Reply:
`window.navigator.platform` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform), I would stay away of using it.